
    syntax = "proto3";

    package nfs;

    message nfsObject 
    {
      int32 methd_identfier = 1 ;
      int32  sys_call = 2 ;  	
      string item_name = 3 ;
      string item_path = 4 ;
      string item_sys_call = 5 ;
      int64  mode_t = 6;
      int64  st_rdev = 7;
      string name_from = 8;
      string name_to = 9;
      int64 st_size = 10;    
      int64  off_set = 11 ;
    }

    message nfsStat
    { 
    	int64 st_dev  = 1;
    	int64 st_ino  = 2;    
    	int64 st_mode = 3;    
    	int64 st_nlink= 4;   
    	int64 st_uid  = 5;   
    	int64 st_gid  = 6;     
    	int64 st_rdev = 7;    
    	int64 st_size = 8;    
    	int64 st_blksize = 9; 
    	int64 st_blocks = 10;  
    	int64 mtime  = 11;
    	int64 ctime  = 12;
    	int64 atime  = 13; 
      int32 methd_identfier = 14;
      string item_name = 15 ;
      string buffer_space = 16;
      int64  off_set = 17 ;
    }

    message nfsVFSStat
    { 

      int64 f_bsize  = 1;
      int64 f_frsize = 2;    
      int64 f_blocks = 3;    
      int64 f_bfree  = 4;   
      int64 f_bavail = 5;   
      int64 f_files  = 6;   
      int64 f_ffree  = 7;   
      int64 f_favail =8 ;   
      int64 f_fsid  = 9;     
      int64 f_flag  = 10;    
      int64 f_namemax  = 11; 
      int32 result = 12; 
    }

    message nfsBool
    {
        int32 result = 1;	
        nfsStat nfs_stat = 2;
        string fi_name = 3;
        string fi_dirname = 4;
        string fi_pathname = 5; 	
        int64 fi_offset = 6 ;
        string fi_fs = 7;
        string fi_mount = 8;
        int64 fi_open_flags = 9;
        int64  mode_t = 10;
        int64  st_rdev = 11;
        string buffer_space = 12;
        int64  st_offset = 13;
        int64  st_size = 14;
    }

    message nfsDirList
    {
    	int32 nfs_dir_result = 1;
    	repeated string nfs_dir_list = 2;
    }


